server/adaptors/integrations/integrations_builder.ts

```diff
@@ -157,6 +204,77 @@ export class IntegrationInstanceBuilder {
     });
   }
 
+  addMDSReference(
+    assets: SavedObject[],
+    dataSource?: string,
+    dataSourceMDSId?: string,
+    dataSourceMDSLabel?: string
+  ): SavedObject[] {
+    if (!dataSource) {
+      return assets;
+    }
+    return assets.map((asset) => {
+      // Check if the asset type is 'index-pattern' or if the title contains 'Timeline'
+      if (
+        asset?.type &&
+        asset?.attributes?.title &&
+        (asset.type === 'index-pattern' ||
+          (asset.type === 'visualization' && asset.attributes.title.includes('Timeline')))
```

Comment:
`asset.attributes.title.includes('Timeline')))` May not be the right check. 
Should check if  `asset.attributes.visState.type ==="timelion"`